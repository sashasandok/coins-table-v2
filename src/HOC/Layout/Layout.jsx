// react
import React, { Component } from 'react'
import PropTypes from 'prop-types'

// styles
import './Layout.css'
import img from '../../assets/images/coin.jpeg'

// semantic-ui
import { Image } from 'semantic-ui-react'

export default class Layout extends Component {
  render() {
    return (
      <div className="layout">
        <header className="layout-header">
          <p>Coins Table</p>
          <div>
            <Image src={img} alt="image" width={50} height={50} />
          </div>
        </header>
        <section className="layout-content">{this.props.children}</section>
        <footer className="layout-footer">&copy; Alex_Sandok 2018</footer>
      </div>
    )
  }
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}
