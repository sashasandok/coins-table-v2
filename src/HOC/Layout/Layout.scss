.layout {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-height: 100vh;
}

.layout-header {
  background-color: #282c34;
  min-height: 60px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  font-size: calc(10px + 2vmin);
  color: white;
}

.layout-header div {
  border-radius: 50%;
}

.layout-header p {
  margin-top: 20px;
}

.layout-header p:hover {
  color: rgb(197, 119, 66);
  text-decoration: underline;
  cursor: pointer;
}

.layout-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  flex: 1 0 auto;
  background-color: rgb(58, 58, 58);
  font-size: calc(10px + 2vmin);
  overflow: auto;
}

.layout-footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #282c34;
  min-height: 50px;
  font-size: calc(10px + 1vmin);
  color: white;
}
