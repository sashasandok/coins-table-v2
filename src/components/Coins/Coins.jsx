// react
import React, { Component } from 'react'
import PropTypes from 'prop-types'

// semantic-ui
import { Table } from 'semantic-ui-react'

// components
import Coin from '../Coin/Coin'

class Coins extends Component {
  render() {
    const { coins, page, exchange } = this.props

    return (
      <Table.Body>
        {coins.map((item, index) => (
          <Coin
            index={index}
            key={item.id}
            coin={item}
            page={page}
            exchange={exchange}
          />
        ))}
      </Table.Body>
    )
  }
}

Coins.propTypes = {
  page: PropTypes.number.isRequired,
  exchange: PropTypes.instanceOf(Object),
  coins: PropTypes.instanceOf(Array),
}

export default Coins
