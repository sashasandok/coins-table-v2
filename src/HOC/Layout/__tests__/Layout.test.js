import React from 'react'
import Layout from '../Layout'
import Coins from '../../../components/Coins/Coins'
import { shallow } from 'enzyme'
import img from '../../../assets/images/coin.jpeg'
import { Image, Table } from 'semantic-ui-react'
import renderer from 'react-test-renderer'

describe('Layout', () => {
  it('testing layout component', () => {
    const props = {
      coins: [],
      page: 1,
      exchange: {},
    }
    const children = (
      <Table>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell width={2}>
              <p>Order Number</p>
            </Table.HeaderCell>
            <Table.HeaderCell width={2}>
              <p>Coin Icon</p>
            </Table.HeaderCell>
            <Table.HeaderCell>
              <p>Coin Name</p>
            </Table.HeaderCell>
            <Table.HeaderCell>
              <p>USD</p>
            </Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Coins
          coins={props.coins}
          page={props.page}
          exchange={props.exchange}
        />
      </Table>
    )

    const tree = renderer
      .create(
        <div className="layout">
          <header className="layout-header">
            <p>Coins Table</p>
            <div>
              <Image src={img} alt="image" width={50} height={50} />
            </div>
          </header>
          <section className="layout-content">{children}</section>
          <footer className="layout-footer">&copy; Alex_Sandok 2018</footer>
        </div>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
    console.log(tree)
  })

  it('testing layout header', () => {
    const component = shallow(<Layout />)
    expect(component.find('.layout-header').children()).toHaveLength(2)
  })

  it('testing layout content', () => {
    const component = shallow(<Layout />)
    expect(component.find('.layout-content').children()).not.toBeNull()
  })

  it('testing layout footer', () => {
    const component = shallow(<Layout />)
    expect(component.find('.layout-footer').text()).toEqual(
      'Â© Alex_Sandok 2018',
    )
  })
})
