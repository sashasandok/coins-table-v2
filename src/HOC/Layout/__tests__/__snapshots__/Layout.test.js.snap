// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout testing layout header 1`] = `
<div
  className="layout"
>
  <header
    className="layout-header"
  >
    <p>
      Coins Table
    </p>
    <div>
      <img
        alt="image"
        className="ui image"
        height={50}
        src="coin.jpeg"
        width={50}
      />
    </div>
  </header>
  <section
    className="layout-content"
  />
  <footer
    className="layout-footer"
  >
    Â© Alex_Sandok 2018
  </footer>
</div>
`;
